File: test-automation-core/src/main/java/com/nortal/test/core/model/ScenarioContext.java
package com.nortal.test.core.model;

import java.time.OffsetDateTime;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import com.nortal.test.core.rest.HttpResponse;
import lombok.Data;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
public class ScenarioContext {
	private final String scenarioId = UUID.randomUUID().toString();
	private final OffsetDateTime scenarioStartTime = OffsetDateTime.now();

	private final ScenarioCommonContext common = new ScenarioCommonContext();

	@Data
	public static class ScenarioCommonContext {
		private HttpResponse<?> lastHttpResponse;

		private final Map<String, String> headers = new HashMap<>();

		public void addHeader(final String key, final String value) {
			headers.put(key, value);
		}
	}


}

File: test-automation-core/src/main/kotlin/com/nortal/test/core/model/ScenarioContext.kt
package com.nortal.test.core.model

import java.time.OffsetDateTime
import java.util.*


class ScenarioContext {
    private val scenarioId = UUID.randomUUID().toString()
    private val scenarioStartTime = OffsetDateTime.now()
    private val common = ScenarioCommonContext()


    class ScenarioCommonContext {
//        private val lastHttpResponse: HttpResponse<*>? = null
        private val headers: MutableMap<String, String> = HashMap()
        fun addHeader(key: String, value: String) {
            headers[key] = value
        }
    }
}